marketplace/netlify-build/src/netlify-app-config.js

```diff
@@ -0,0 +1,254 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { uniq, uniqBy } from 'lodash-es';
+
+import {
+  Notification,
+  Spinner,
+  FieldGroup,
+  CheckboxField,
+  TextLink
+} from '@contentful/forma-36-react-components';
+import { Workbench } from '@contentful/forma-36-react-components/dist/alpha';
+
+import { parametersToConfig, configToParameters } from './config';
+import {
+  currentStateToEnabledContentTypes,
+  enabledContentTypesToTargetState
+} from './target-state';
+
+import NetlifyConnection from './netlify-connection';
+import NetlifyConfigEditor from './netlify-config-editor';
+import * as NetlifyClient from './netlify-client';
+import * as NetlifyIntegration from './netlify-integration';
+
+import styles from './styles';
+
+export default class NetlifyAppConfig extends React.Component {
+  static propTypes = {
+    sdk: PropTypes.object.isRequired
+  };
+
+  async componentDidMount() {
+    this.init();
+  }
+
+  componenWillUnmount() {
+    this.stopPolling();
+  }
+
+  state = { ready: false };
+
+  init = async () => {
+    const { sdk } = this.props;
+    const { app } = sdk.platformAlpha;
+
+    app.onConfigure(this.onAppConfigure);
+
+    const [parameters, currentState, contentTypesResponse] = await Promise.all([
+      app.getParameters(),
+      app.getCurrentState(),
```

Comment:
Non-blocking: I'm starting to wonder if these values should actually get passed into the callback `app.onConfigure` takes for ease-of-use.

Reply:
I should clarify: Not saying to remove them from the app API, just wondering if we can also pass them into the callback as arguments that way you can have a sync callback if those are the only values you need.

Reply:
Computing current state can be HTTP intensive. If you don't need it but we still pass it to the callback we waste time.

Reply:
Makes sense.